extension/encoding/awscloudwatchmetricstreamsencodingextension/extension_test.go

```diff
@@ -17,7 +17,7 @@ func TestNew_JSON(t *testing.T) {
 	require.NotNil(t, e)
 
 	_, err = e.UnmarshalMetrics([]byte{})
-	assert.EqualError(t, err, `failed to unmarshal metrics as 'json' format: 0 metrics were extracted from the record`)
+	assert.NoError(t, err)
```

Comment:
How about we input some invalid JSON, so it's clear by looking at the error that the JSON unmarshaler is reached?